### PR TITLE
830 singleton for util classes

### LINKED ISSUE #830: Use `Singleton Design pattern` for Utils classes
Utils classes are present at: (ending with *utils.java)
https://github.com/Swati4star/Images-to-PDF/tree/master/app/src/main/java/swati4star/createpdf/util

For reference:
Use method 3 Lazy initialization from below link for implementation:
https://www.geeksforgeeks.org/java-singleton-design-pattern-practices-examples/

### PR DESCRIPTION
**Pull request by sebastianjooo**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.
If there are any UI change, please include the screenshots also.

Fixes #830

Used Singleton Desing Pattern for following Utils classes:
-ImageEnhancementOptionsUtils
-ImageFilterUtils
-ImageSortUtils
-ImageUtils
-MergePdfEnhancementOptionsUtils
-PermissionsUtils
-RecentUtil
-ResultUtils
-StringUtils
-TextEnhancementOptionsUtils
-ThemeUtils
-WhatsNewUtils

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x ] `./gradlew assembleDebug assembleRelease`
- [x ] `./gradlew checkstyle`

# Checklist:
- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings

### CODE REVIEW

**Comment by Swati4star** [comment]
@sebastianjooo Please resolve the merge conflicts.

Also, fill in the issue template & mention the classes for which you have implemented the singleton pattern.

**Comment by sebastianjooo** [comment]
yes, i'll do

**Comment by sebastianjooo** [comment]
Should I close this pull request and open a new one, or can i update this one?

**Comment by sebastianjooo** [comment]
i updated the issue template at the top  @Swati4star

File: app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java [hunk 1]
```diff
@@ -0,0 +1,529 @@
+package swati4star.createpdf.fragment;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java:1]: This file should not be added. Remove this file

File: build.gradle [hunk 2]
```diff
@@ -9,7 +9,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.1'
+        classpath 'com.android.tools.build:gradle:3.6.3'
```
Comment (Swati4star) [inline build.gradle:12]: remove the build version related changes also.
I'll update these later

File: gradle.properties [hunk 3]
```diff
@@ -17,4 +17,6 @@ org.gradle.jvmargs=-Xmx4096m -XX:MaxPermSize=4096m -XX:+HeapDumpOnOutOfMemoryErr
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
 android.enableR8=true
-android.enableSeparateAnnotationProcessing=true
\ No newline at end of file
+android.enableSeparateAnnotationProcessing=true
+
+android.useAndroidX=true
```
Comment (Swati4star) [inline gradle.properties:22]: is it required?

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Sat Oct 12 14:58:30 IST 2019
+#Sun Apr 19 13:26:32 CEST 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
```
Comment (Swati4star) [inline gradle/wrapper/gradle-wrapper.properties:6]: revert this

File: gradle.properties (hunk 3 above)
Comment (Swati4star) [reply gradle.properties:22]: @sebastianjooo This is left

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#Sat Oct 12 14:58:30 IST 2019
+#Sun Apr 19 13:26:32 CEST 2020
```
Comment (Swati4star) [inline gradle/wrapper/gradle-wrapper.properties:1]: you can revert this file also.

**Comment by Swati4star** [comment]
Great @sebastianjooo 👍

**Pull request merged**